### PR TITLE
Added new actions and bugfixes

### PR DESCRIPTION
**Pull request by cs-sameer-pathan**
#### Changes:
- Added following new actions:
  - Get File
  - Delete File
  - Search Code
- Updated output schema of `Push Changes` action.

#### Fix:
- Fixed a bug where `Clone Repository` action fails with error message `File is not a zip file` for any issue.
- Fixed a bug where `Push Changes` action not deleting files on GitHub if deleted locally.
- Fixed a bug where `Update Remote Repository` action not deleting files from cloned repository if removed from given file.

#### UTCs:
- [x] Tested and verified `Clone Repository` action.
- [x] Tested and verified `Push Changes` action.
- [x] Tested and verified `Update Remote Repository` action.
- [x] Tested and verified `Get File` action.
- [x] Tested and verified `Delete File` action.
- [x] Tested and verified `Search Code` action.

### CODE REVIEW

File: github/info.json [hunk 1]
```diff
@@ -7372,6 +7597,156 @@
...
+            }
+          }
+        ],
+        "total_count": "",
+        "incomplete_results": ""
+      },
+      "parameters": [
+        {
+          "title": "Search Query",
+          "name": "q",
```
Comment (cs-amol-phatangade) [inline github/info.json]: Please check if you can update this parameter name with some meaning full name for understanding purpose.

File: github/operations.py [hunk 2]
```diff
@@ -583,6 +622,44 @@ def get_web_url(config, params, *args, **kwargs):
...
+            decoded_bytes = base64.b64decode(response.get('content'))
+            response['content'] = decoded_bytes.decode('utf-8')
+        except Exception as err:
+            logger.error(err)
+    return response
+
+
+def delete_file_from_repository(config, params, *args, **kwargs):
+    github = GitHub(config)
+    payload = {
```
Comment (cs-amol-phatangade) [inline github/operations.py:641]: You can use same logic here:
payload = {k: v for k, v in params.items() if
               v is not None and v != '' and v != {} and v != []}

**Review by cs-amol-phatangade — APPROVED** [review]

File: github/operations.py (hunk 2 above)
Comment (cs-sameer-pathan) [reply github/operations.py:641]: As all parameters are required or provided , no need to add this logic

File: github/info.json (hunk 1 above)
Comment (cs-sameer-pathan) [reply github/info.json]: Done

**Pull request merged**